src/Dscp/Crypto/Encrypt.hs

```diff
@@ -0,0 +1,116 @@
+
+-- | Utilities for encrypting/decrypting byte arrays
+
+module Dscp.Crypto.Encrypt
+       ( -- * Passphrases
+         PassPhrase
+       , getPassPhrase
+       , mkPassPhrase
+       , minPassPhraseLength
+
+         -- * Encrypted bytearray
+       , Encrypted
+       , eAuthTag
+       , eCiphertext
+
+         -- * Utility functions
+       , encrypt
+       , decrypt
+       ) where
+
+import Crypto.Cipher.AES (AES256)
+import Crypto.Cipher.Types (AEAD, AEADMode (AEAD_GCM), AuthTag, BlockCipher (aeadInit),
+                            Cipher (cipherInit), IV, aeadSimpleDecrypt, aeadSimpleEncrypt, nullIV)
+import Crypto.Error (onCryptoFailure)
+import Data.ByteArray (ByteArray, ByteArrayAccess)
+import qualified Data.ByteArray as BA
+import Fmt ((+|), (|+))
+
+import Dscp.Crypto.Impl (hash)
+
+-------------------------------------------------------------
+-- Passphrases
+-------------------------------------------------------------
+
+newtype PassPhrase = PassPhrase
+    { getPassPhrase :: ByteString
+    } deriving (Eq, Ord, Show, Monoid, ByteArray, ByteArrayAccess)
```

Comment:
Don't we want blind `Show` instance?

Reply:
Fair enough. I'll actually make blind `Buildable` instance too.